### PR TITLE
Moving to Clopper-Pearson uncertainties in efficiency histograms

### PR DESCRIPTION
**Pull request by natalia-korneeva**
As decided at BTV@HLT meeting and inspired by https://github.com/cms-sw/cmssw/pull/9863/ we replace binomial uncertainties by Clopper-Pearson ones in efficiency histograms. Clopper-Perason uncertainties provide better description in low efficiency cases.

### CODE REVIEW

**Comment by mtosi** [comment]
@cmsbuild, please test

File: HLTriggerOffline/Btag/src/HLTBTagHarvestingAnalyzer.cc [hunk 1]
```diff
@@ -301,7 +301,8 @@ TH1F  HLTBTagHarvestingAnalyzer::calculateEfficiency1D(DQMStore::IBooker& ibooke
 		double e;
 		if(d!=0){
 			e=n/d;
-			err =  sqrt(e*(1-e)/d); //from binomial standard deviation
+			err = std::max(e - TEfficiency::ClopperPearson(d, n, 0.683, false), TEfficiency::ClopperPearson(d, n, 0.683, true) - e);
```
Comment (mtosi) [inline HLTriggerOffline/Btag/src/HLTBTagHarvestingAnalyzer.cc]: naive question
why don't you make use of the DQMGenericClient for creating the efficiency plots ?

Comment (natalia-korneeva) [reply HLTriggerOffline/Btag/src/HLTBTagHarvestingAnalyzer.cc]: Are you commenting this particular change or the whole calculateEfficiency1D function?
Could you please point me to examples of creating efficiency histograms with the DQMGenericClient?

**Comment by mtosi** [comment]
in general, of course :P

you can see the client used by BTV ;)
https://github.com/natalia-korneeva/cmssw/blob/3cf722bc3a641cee3d73d3abf6d7c1c70c5aec5b/DQMOffline/Trigger/python/BTaggingMonitoring_Client_cff.py

**Comment by defranchis** [comment]
Hi Mia, can you actually create efficiency plots using Clopper-Pearson uncertainties using the DQMGenericClient? Anyway I would say that the HLTriggerOffline/Btag package is a well-working self-consistent code, therefore we can consider the changes you propose as possible optimization for the future - at the moment we are spending all the manpower we have for more urgent developments. Would you agree with that?

**Comment by mtosi** [comment]
ciao

not sure it is the case, as now
but one can always improves the code used by the collaboration ;)

  mia

On Mon, Oct 16, 2017 at 4:53 PM, Matteo Defranchis <notifications@github.com

**Comment by imarches** [comment]
Dear all,

thanks for your inputs. I would suggest to keep the PR within its current scope and merge it as it is (unless there are specific comments to the code changes). We have assigned a new person to BTV HLT DQM, @HeinerTholen. Given his marked software skills, I'm sure he will give important and more structural updates in the coming weeks.

Cheers,

Ivan

**Comment by HeinerTholen** [comment]
Hey there,

@defranchis: the DQMGenericClient uses Clopper-Pearson (at least for ROOT_VERSION_CODE >= ROOT_VERSION(5,27,0)):
https://github.com/cms-sw/cmssw/blob/09c3fce6626f70fd04223e7dacebf0b485f73f54/DQMServices/ClientConfig/plugins/DQMGenericClient.cc#L486-L491

@imarches: I'll keep this in mind and see if it can be simplified easily. There are a couple of things I have to learn about DQM first.. ^^

Cheers, H

**Comment by dmitrijus** [comment]
+1

> There are a couple of things I have to learn about DQM first.. ^^
Please email me if you have any questions :)

**Comment by davidlange6** [comment]
merge

**Pull request merged**